### ISSUE TITLE
TypeError: Couldn't build proto file into descriptor pool! Invalid proto descriptor for file "blobopera/libretto_blobopera.libretto.fragment.proto":

### ISSUE THREAD

**Issue by Alee14** [0x2b3bfa0/python-blobopera#9]
Hello so when I ran this command `blobopera recording import Sunset\ Paradise\ Season\ 1\ -\ Official\ Opening\ Piano.musicxml Sunset\ Paradise\ Season\ 1\ -\ Official\ Opening\ Piano.out`, it gave me this error

```python
Traceback (most recent call last):
  File "/usr/bin/blobopera", line 5, in <module>
    from blobopera.command import application
  File "/home/andrew/.local/lib/python3.9/site-packages/blobopera/command/__init__.py", line 6, in <module>
    from . import jitter, libretto, recording
  File "/home/andrew/.local/lib/python3.9/site-packages/blobopera/command/libretto.py", line 5, in <module>
    from ..libretto import Corpus
  File "/home/andrew/.local/lib/python3.9/site-packages/blobopera/libretto.py", line 38, in <module>
    class Fragment(proto.Message):
  File "/home/andrew/.local/lib/python3.9/site-packages/proto/message.py", line 263, in __new__
    file_info.generate_file_pb(new_class=cls, fallback_salt=full_name)
  File "/home/andrew/.local/lib/python3.9/site-packages/proto/_file_info.py", line 102, in generate_file_pb
    pool.Add(self.descriptor)
TypeError: Couldn't build proto file into descriptor pool!
Invalid proto descriptor for file "blobopera/libretto_blobopera.libretto.fragment.proto":
  blobopera.libretto.Fragment.phonemes: Fields of oneofs must themselves have label LABEL_OPTIONAL.
```

**Comment by Alee14** [comment]
Fixed by compiling it manually

**Comment by fjferdiez** [comment]
Hello! I ran into this problem, may I ask how did you complied it?

**Comment by Alee14** [comment]
I followed the contributing guide

**Comment by 0x2b3bfa0** [comment]
@Alee14 & @fjferdiez, could any of you try to reproduce the issue and post the output of the `pip freeze` command? I have been unable to reproduce with neither of the released versions. 🤔

**Comment by HAL9000Thomas** [comment]
@0x2b3bfa0 Got the same issue. The command output of pip freeze is:

```
abc@abc blob % blobopera recording import Adeste_Fideles_TTBB.mxl output.binary
Traceback (most recent call last):
  File "/usr/local/bin/blobopera", line 5, in <module>
    from blobopera.command import application
  File "/usr/local/lib/python3.9/site-packages/blobopera/command/__init__.py", line 6, in <module>
    from . import jitter, libretto, recording
  File "/usr/local/lib/python3.9/site-packages/blobopera/command/libretto.py", line 5, in <module>
    from ..libretto import Corpus
  File "/usr/local/lib/python3.9/site-packages/blobopera/libretto.py", line 38, in <module>
    class Fragment(proto.Message):
  File "/usr/local/lib/python3.9/site-packages/proto/message.py", line 263, in __new__
    file_info.generate_file_pb(new_class=cls, fallback_salt=full_name)
  File "/usr/local/lib/python3.9/site-packages/proto/_file_info.py", line 102, in generate_file_pb
    pool.Add(self.descriptor)
TypeError: Couldn't build proto file into descriptor pool!
Invalid proto descriptor for file "blobopera/libretto_blobopera.libretto.fragment.proto":
  blobopera.libretto.Fragment.phonemes: Fields of oneofs must themselves have label LABEL_OPTIONAL.

abc@abc blob % pip freeze
blobopera==1.0.1
certifi==2021.5.30
chardet==4.0.0
click==7.1.2
colorama==0.4.4
cycler==0.10.0
idna==2.10
joblib==1.0.1
kiwisolver==1.3.1
matplotlib==3.4.2
more-itertools==8.8.0
music21==6.7.1
numpy==1.21.0
Pillow==8.2.0
proto-plus==1.19.0
protobuf==3.17.3
pyparsing==2.4.7
python-dateutil==2.8.1
requests==2.25.1
shellingham==1.4.0
six==1.16.0
typer==0.3.2
urllib3==1.26.6
webcolors==1.11.1
```


Is there any workaround for solving this issue?

**Comment by HAL9000Thomas** [comment]
Ok, following the contributing guide worked for me too. Make sure to have poetry pre-installed and run the actual plobopera command in this way in the root folder of the cloned project: `poetry run blobopera recording import myfile.mxl output.binary`

**Comment by 0x2b3bfa0** [comment]
Follow-up on https://github.com/googleapis/proto-plus-python/issues/238

**Comment by 0x2b3bfa0** [comment]
Fixed with a8b5ab537cb85de193c4f70eec9506a06f34cbe5

**Comment by 0x2b3bfa0** [comment]
@Alee14, @fjferdiez & @HAL9000Thomas, sorry for the issue. It has been fixed in 1.0.2 by pinning the `protobuf` package to an older, working version. Running `pip install --upgrade blobopera` should be enough to fix the issue on your systems.
